executor/runtime/docker/docker.go

```diff
@@ -1175,16 +1176,36 @@ func (r *DockerRuntime) Prepare(ctx context.Context, pod *v1.Pod) error { // nol
 
 	containerCreateBody, err = r.client.ContainerCreate(ctx, dockerCfg, hostCfg, nil, r.c.TaskID())
 	r.c.SetID(containerCreateBody.ID)
-
-	r.metrics.Timer("titus.executor.dockerCreateTime", time.Since(dockerCreateStartTime), r.c.ImageTagForMetrics())
 	if docker.IsErrNotFound(err) {
 		return &runtimeTypes.RegistryImageNotFoundError{Reason: err}
 	}
 	if err != nil {
 		goto error
 	}
 	ctx = logger.WithField(ctx, "containerID", r.c.ID())
-	logger.G(ctx).Info("Container successfully created")
+	logger.G(ctx).Info("Main Container successfully created")
+
+	if len(pod.Spec.Containers) > 1 {
+		mainContainerRoot, err = r.inspectAndGetMainContainerRoot(ctx)
+		if err != nil {
+			goto error
```

Comment:
A more idiomatic pattern for this:

```go
func (r *DockerRuntime) Prepare(ctx context.Context, pod *v1.Pod) (retErr error) {

        ...

        defer func() {
                if retErr != nil && !docker.IsErrNotFound(retErr) { // if the difference in behavior for the "not found" error is intentional
                        tracehelpers.SetStatus(retErr, span)
                        log.Error("Unable to create main container: ", retErr)
                        r.metrics.Counter("titus.executor.dockerCreateContainerError", 1, nil)
                }
        }()
```

Then just return errors as needed.

Reply:
I don't know about the history behind the `goto` thing.
I agree I find it a bit surprising.

Let me propose this on a different PR.